apps/prairielearn/src/migrations/20231003211522_lti13_instance__create.sql

```diff
@@ -0,0 +1,65 @@
+INSERT INTO
+  authn_providers
+VALUES
+  (6, 'LTI 1.3')
+ON CONFLICT (id) DO NOTHING;
+
+CREATE TABLE IF NOT EXISTS
+  pl_lti13_instances (
+    id bigserial PRIMARY KEY,
+    institution_id BIGINT NOT NULL REFERENCES institutions ON UPDATE CASCADE ON DELETE SET NULL,
+    created_at timestamptz NOT NULL DEFAULT current_timestamp,
+    deleted_at timestamptz,
+    platform text,
+    name text,
+    tool_platform_name text,
+    keystore jsonb,
+    issuer_params jsonb,
+    client_params jsonb,
+    custom_params jsonb,
+    uid_attribute text,
+    uin_attribute text,
+    name_attribute text,
+    access_tokenset jsonb,
+    access_token_expires_at timestamptz
+  );
+
+CREATE INDEX IF NOT EXISTS pl_lti13_instances_institution_id_idx ON pl_lti13_instances (institution_id);
+
+CREATE TABLE IF NOT EXISTS
+  pl_lti13_platform_defaults (
+    id bigserial PRIMARY KEY,
+    platform text NOT NULL DEFAULT 'Unknown',
+    issuer_params jsonb,
+    client_params jsonb,
+    custom_params jsonb,
+    display_order integer
+  );
+
+INSERT INTO
+  pl_lti13_platform_defaults (
+    platform,
+    issuer_params,
+    client_params,
+    display_order
+  )
+VALUES
+  ('Unknown', null, null, 0),
+  (
+    'Canvas Production',
+    '{
+   "issuer": "https://canvas.instructure.com",
+   "jwks_uri": "https://sso.canvaslms.com/api/lti/security/jwks",
+   "token_endpoint": "https://sso.canvaslms.com/login/oauth2/token",
+   "authorization_endpoint": "https://sso.canvaslms.com/api/lti/authorize_redirect"
+}',
+    '{
+   "client_id": "10000000000001",
+   "redirect_uris": [
+      "https://pl.tbkenny.com/pl/lti13_instance/1/auth/callback"
+   ],
+   "token_endpoint_auth_method": "private_key_jwt",
+   "token_endpoint_auth_signing_alg": "RS256"
+}',
+    10
+  );
```

Comment:
I figured the list of default configs would grow as we supported more platforms and it seemed more flexible to add various ones (or dev/site specific ones like my tbkenny environment) into the database instead of the code base.

This table is only used to populate the options the javascript asset uses to load a different platform default into the `institutionAdminLti13` page.

But I don't have a strong opinion on this either way if you prefer it not be a table.

Reply:
Maybe things will shake out differently in practice, but I can't see why we'd want to add custom defaults in one production environment and not all the rest.

If you really want this to be configurable for local dev, consider defining extra ones in the config file. In production, if we did have environment-specific things, we'd want them in configs anyways so that they can be version-controlled (unlike one-off manually-run database migrations).

Reply:
I agree we don't want this table in the DB.

Reply:
Moved to config in 875a68a2a